labgrid/config.py

```diff
@@ -153,6 +153,64 @@ def set_option(self, name, value):
         assert isinstance(value, str)
         self.data.setdefault('options', {})[name] = value
 
+    def get_target_option(self, target, name, default=None):
+        """Retrieve an entry from the options subkey under the specified target
+           subkey
+
+        Args:
+            target (str): name of the target
+            name (str): name of the option
+            default (str): A default parameter in case the option can not be
+                found
+
+        Returns:
+            str: value of the option or default parameter
+
+        Raises:
+            KeyError: if the requested key can not be found in the
+                configuration, or if the target can not be found in the
+                configuration.
+        """
+        if target not in self.data['targets']:
+            raise KeyError("No target '{}' found in configuration".format(target))
+
+        try:
+            return str(self.data['targets'][target]['options'][name])
+        except (KeyError, TypeError):
+            # Empty target declarations become None in the target dict, hence
+            # TypeError when we try to subscript it.
+            if default is None:
+                raise KeyError("No option '{}' found in configuration for target '{}'".format(name, target))
+            else:
+                return default
+
+    def set_target_option(self, target, name, value):
+        """Set an entry in the options subkey under the specified target subkey
+
+        Args:
+            target (str): name of the target
+            name (str): name of the option
+            value (str): the new value
+
+        Raises:
+            KeyError: if the requested target can not be found in the
+                configuration
+        """
+        assert isinstance(target, str)
+        assert isinstance(name, str)
+        assert isinstance(value, str)
```

Comment:
I don't think its necessary to restrict options to strings here, since the YAML syntax also allows integers, dictionaries or other data types.

Reply:
Fair point, although they are stringified later as you pointed out. We could change so that both per-target options and global options may hold arbitrary value types.